lib/connect/index.js

```diff
@@ -60,23 +58,32 @@ function connect (brokerUrl, opts) {
   opts = opts || {}
 
   if (brokerUrl) {
-    var parsed = url.parse(brokerUrl, true)
-    if (parsed.port != null) {
-      parsed.port = Number(parsed.port)
-    }
-
-    opts = xtend(parsed, opts)
-
     if (opts.protocol === null) {
       throw new Error('Missing protocol')
     }
+    var parsed = new URL(brokerUrl)
+    // the URL object is a bit special, so copy individual
+    // items to the opts object
+    opts.hostname = parsed.hostname
+    opts.host = parsed.host
+    opts.protocol = parsed.protocol
+    opts.port = Number(parsed.port) || null
+    opts.username = parsed.username
+    opts.password = parsed.password
+    opts.searchParams = parsed.searchParams
```

Comment:
I am pretty sure we don't need more. And the tests agreed with me on that ;-)

This is what  url.parse returns:
```javascript
> url.parse("https://user:password@www.example.com:8080/?a=b")
Url {
  protocol: 'https:',
  slashes: true,
  auth: 'user:password',
  host: 'www.example.com:8080',
  port: '8080',
  hostname: 'www.example.com',
  hash: null,
  search: '?a=b',
  query: 'a=b',
  pathname: '/',
  path: '/?a=b',
  href: 'https://user:password@www.example.com:8080/?a=b' 
``` 

Kind regards,
Hans

Reply:
BTW:  @nosovk already approved on September 2  so its not just me ;-)

Reply:
It should actually be `opts.host = parsed.hostname`
This project uses `host` as the name for `hostname` (just domain, without port)

Reply:
Except for the 11 locations where opts.hostname is being used ;-)
https://github.com/mqttjs/MQTT.js/search?q=opts.hostname

and currently (without my PR) the results from url.parse are copied into opts.

So currently:
- `opts.host` is domain with port (www.example.com:8080) 
- `opts.hostname` is domain without port  (www.example.com)

The comments in type definitions seem to be a bit ahead of the code ;-)

But anyway:  

For me its just a hobby, and I saw the issue and thought I'd help out.
If you all feel it can be done way better than I did, feel free! 
(its only a few lines of code and from my PR you know where to look ;-))
That would save me the effort of rebasing (again) as well.

If you still want me to rebase because you are all way too busy, just let me know when its safe to do so I don't need to rebase a third time ;-)

Reply:
appreciate your contributions just doing due diligence :)

Reply:
Well, in that case: #1148 is also still open ;-)
It would convenient for me if that could be either merged or fixed in some other way better way :-)
(e.g. generate fresh certs if not available etc )